src/scripts/main.js

```diff
@@ -1,3 +1,267 @@
 'use strict';
 
 // write your code here
+const body = document.querySelector('body');
+
+const rows = body.querySelectorAll('tr');
+const button = document.querySelector('button');
+const tableScore = body.querySelector('.game-score');
+const messages = body.querySelectorAll('.message');
+const winMessage = body.querySelector('.message-win');
+const loseMessage = body.querySelector('.message-lose');
+const cells = body.querySelectorAll('.field-cell');
+
+const board = [
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+];
+const newNumber = [2, 2, 2, 2, 2, 2, 2, 2, 4, 2];
+let gameEnded = false;
+
+button.addEventListener('click', (e) => {
+  if (!button.classList.contains('restart')) {
+    button.classList.remove('start');
+    button.innerText = 'Restart';
+    button.classList.add('restart');
+  }
+
+  for (let r = 0; r < board.length; r++) {
+    for (let c = 0; c < board.length; c++) {
+      board[r][c] = 0;
+
+      const tile = rows[r].children[c];
+
+      updateTile(tile, 0);
+    }
+  }
+
+  [...messages].map(message => message.classList.add('hidden'));
```

Comment:
Use `forEach` instead of `map` to iterate over the messages, as you are not creating a new array

Reply:
+